src/commands/documentation/api.js

```diff
@@ -0,0 +1,190 @@
+import { Command } from 'discord.js-commando'
+import { getAPI, findAPI, APINotFoundError } from '../../services/api'
+import { RichEmbed } from 'discord.js'
+import { EMPTY_MESSAGE } from '../../utils/constants'
+import { inlineCode, blockCode } from '../../utils/string'
+import { tryDelete } from '../../utils/messages'
+import { DEFAULT_EMBED_COLOUR } from '../../utils/embed'
+
+module.exports = class DocumentationAPICommand extends Command {
+  constructor(client) {
+    super(client, {
+      args: [
+        {
+          key: 'lookup',
+          type: 'string',
+          prompt: 'what do you want to look up in the API docs?',
+        },
+      ],
+      name: 'api',
+      examples: [
+        inlineCode('!api errorHandler'),
+        inlineCode('!api Vue.observable'),
+        inlineCode('!api observable'),
+        inlineCode('!api computed'),
+        inlineCode('!api template'),
+        inlineCode('!api mounted'),
+        inlineCode('!api directives'),
+        inlineCode('!api mixins'),
+        inlineCode('!api delimiters'),
+        inlineCode('!api vm.$props'),
+        inlineCode('!api vm.$watch'),
+        inlineCode('!api vm.$on'),
+        inlineCode('!api vm.$mount'),
+        inlineCode('!api $props'),
+        inlineCode('!api $watch'),
+        inlineCode('!api $on'),
+        inlineCode('!api $mount'),
+        inlineCode('!api v-html'),
+        inlineCode('!api ref'),
+        inlineCode('!api keep-alive'),
+      ],
+      group: 'documentation',
+      aliases: ['a'],
+      guildOnly: false,
+      memberName: 'api',
+      description: 'Look something up in the API docs.',
+    })
+  }
+
+  hasPermission() {
+    return true
+  }
+
+  async run(msg, args) {
+    const { lookup } = args
+
+    let embed,
+      isDisambiguation = false
+
+    try {
+      // Try to find an exact match (or alias).
+      let api = getAPI(lookup)
+
+      if (api) {
+        embed = this.buildResponseEmbed(msg, api)
+      } else {
+        // Attempt a fuzzy search.
+        api = findAPI(lookup)
+
+        if (api) {
+          if (api.length > 1) {
+            embed = this.buildDisambiguationEmbed(msg, lookup, api)
+            isDisambiguation = true
+          } else if (api.length === 1) {
+            embed = this.buildResponseEmbed(msg, api[0])
+          }
+        }
+      }
+
+      if (!api || api.length === 0) {
+        throw new APINotFoundError(
+          `Could not find anything in the API matching "${lookup}".`
+        )
+      }
+
+      const reply = await msg.channel.send(EMPTY_MESSAGE, { embed })
+      tryDelete(msg, 7500)
```

Comment:
How did you settle on these values? We should really make these config settings, I think.

Reply:
The values were essentially made up by me on the spot.

I agree but I'd like to tackle this in another PR.

The reason for that being that all the PRs have code like this - it doesn't make sense to implement that in this PR, agreed?

See #38.